code_map.py

```diff
@@ -221,17 +306,20 @@ def navigate_to_line(map_view, give_back_focus=False):
         # navigate only if a valid map node is clicked or has caret
         return
 
-    source_code_view = None
+    if CURRENT_TEMP_ID:
+        source_code_view = TEMP_VIEWS[CURRENT_TEMP_ID]
+    else:
+        source_code_view = None
 
-    if code_map_generator.source:
-        source_code_view = win().find_open_file(
-            code_map_generator.source)
-        if not source_code_view:
-            source_code_view = win().open_file(
+        if code_map_generator.source:
+            source_code_view = win().find_open_file(
                 code_map_generator.source)
+            if not source_code_view:
+                source_code_view = win().open_file(
+                    code_map_generator.source)
 
```

Comment:
CURRENT_TEMP_ID = view.id() when a temporary view is active, otherwise it's None. Iif not None, the source_code_view is stored in the variable and it's taken from there. Being a temporary view, it wouldn't be possible for CodeMap to find it again with other means.